### DISCUSSION TITLE
Branches not in metadata: branches out of range

### DISCUSSION THREAD

**Discussion by ben18785** [ropensci/targets#982] [Help]
I have been really enjoying using targets for a project that me and a wider team of statisticians are on. We have now, however, run into this issue: whenever we run `tar_make()` or `tar_load_globals()` or `tar_manifest()` we get the following error:

> Last error: branches not in metadata: branches out of range.

A google points us only to the original documentation for `tar_read_raw`.

I'm afraid I can't provide a reproducible example since our code is all unable to be shared (we're working on a government project) and I don't know what causes the issue so can't make a simpler version that reproduces the error.

But would someone be able to suggest where we might be able to start looking?

**Comment by wlandau** [comment]
Looks like you are using [dynamic branching](https://books.ropensci.org/targets/dynamic.html), either through the `pattern` argument of `tar_target()` or one of the batched replication functions in [`tarchetypes`](https://docs.ropensci.org/tarchetypes/reference/index.html). It's hard to say exactly what's happening, but it looks like the expected dynamic branches went missing from the data store somehow. If you are working on a single project with a whole team, it is important to use separate per-person [data stores](https://books.ropensci.org/targets/data.html#local-data-store). If multiple people call `tar_make()` on the same project at the same time, or there are multiple instances of `tar_make()` running at the same time on the same project, this can corrupt the data store and potentially cause the problem you mentioned. Otherwise, it's hard to say why it's happening in your case.

**Reply by ben18785** [reply]
Thanks. We aren't using dynamic branching actually (nor anything from `tarchetypes`), which is a bit strange. Is there a way to restart targets and just rerun everything?

**Reply by ben18785** [reply]
I've actually managed to locate the issue: someone on our team had put a `tar_read` statement inside one of the scripts. This was wreaking havoc with things. I wonder whether it might be possible to have a more informative error message?

**Reply by dewoller** [reply]
Thanks @ben18785 muchly, that clue pointed me in the exact right direction.

**Reply by billdenney** [reply]
Thanks for this discussion. I had this issue as well. My issue was that I was running `tar_read()` on a function instead of a target name.
